File: src/main/java/e6eo/finalproject/entity/UsersEntity.java
package e6eo.finalproject.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity(name = "users")
@Data
@NoArgsConstructor
public class UsersEntity {
    @Id
    @Column(name = "user_id", unique = true, nullable = false)
    private String userId;
    @Column(name = "pw", nullable = false)
    private String pw;
    @Column(name = "nickname", nullable = false)
    private String nickName;
    @Column(name = "inner_id", nullable = true)
    private String innerId;
    @Column(name = "refresh_token", nullable = true)
    private String refreshToken;

    @Builder
    public UsersEntity(String userId, String pw, String nickName, String innerId, String refreshToken){
        this.userId = userId;
        this.pw = pw;
        this.nickName = nickName;
        this.innerId = innerId;
        this.refreshToken = refreshToken;

    }
}

File: src/main/kotlin/e6eo/finalproject/entity/UsersEntity.kt
package e6eo.finalproject.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import lombok.Data
import lombok.NoArgsConstructor

@Entity(name = "users")
@Data
@NoArgsConstructor
data class UsersEntity(
    @Id
    @Column(name = "user_id", unique = true, nullable = false)
    var userId: String? = null,

    @Column(name = "pw", nullable = true)
    var pw: String? = null,

    @Column(name = "nickname", nullable = false)
    var nickName: String? = null,

    @Column(name = "observe_token", nullable = true, unique = true)
    var observeToken: String? = null,

    @Column(name = "inner_id", nullable = true)
    var innerId: String? = null,

    @Column(name = "refresh_token", nullable = true)
    var refreshToken: String? = null
)